core/services/gateway/constants.go

```diff
@@ -0,0 +1,9 @@
+package gateway
+
+type ErrorCode int
+
+const (
+	ErrNoError ErrorCode = iota
+	ErrUserMessageParseError
+	ErrFatalError
```

Comment:
This naming pattern refers to predefined errors (the prefix Err) and therefore confusing, this is common to Go.
for example, the standard http package: https://pkg.go.dev/net/http#pkg-variables
Maybe you can simply remove the prefix here.